### PR TITLE
Convert to standard HERE routing function

### PR DESCRIPTION
**Pull request by tankedman**

### CODE REVIEW

**Comment by tankedman** [comment]
For now, the get_links_between_two_nodes function returns the following json:

("source", "target", "links")

Source: the start node
Target: the end node
Links: A list of dictionary objects:  {'link_dir', 'seq', 'segment_id', 'geom', 'length'}

File: backend/app/routes.py [hunk 1]
```diff
@@ -123,36 +123,63 @@ def get_links_between_two_nodes(from_node_id, to_node_id):
...
 
                 SELECT 
-                    %(node_start)s,
-                    %(node_end)s,
-                    array_agg(st_name),
-                    array_agg(link_dir),
-                    ST_AsGeoJSON(ST_union(ST_linemerge(geom))) AS geometry
-                FROM results
-                INNER JOIN here.routing_streets_name on edge = id'''
+					link_dir,
```
Comment (Nate-Wessel) [inline backend/app/routes.py]: @tankedman can you please clean up this code? Remove commented sections and fix the indenting.

Comment (Nate-Wessel) [reply backend/app/routes.py]: Please also add table name prefixes to the selected fields so it's clear where they're coming from.

File: backend/app/routes.py [hunk 2]
```diff
@@ -123,36 +123,63 @@ def get_links_between_two_nodes(from_node_id, to_node_id):
             select_sql = '''
```
Comment (Nate-Wessel) [inline backend/app/routes.py]: The comment above also needs to be updated.

File: backend/app/routes.py [hunk 3]
```diff
@@ -117,42 +117,41 @@ def get_links_between_two_nodes(from_node_id, to_node_id):
 
     with connection:
         with connection.cursor() as cursor:
-            #Uses pg_routing to route between the start node and end node on the HERE
-            #links network. Returns inputs and an array of link_dirs and a unioned line
-            #TODO: convert to here_gis.get_links_between_nodes
+            #Uses here_gis.get_links_btwn_nodes_22_2 to route between the start node and end node on the HERE
+            #links network. Returns an array of link objects.
+            #TODO: create unique pathnames
```
Comment (Nate-Wessel) [inline backend/app/routes.py]: Probably easiest to just do this on the frontend.

File: backend/app/routes.py [hunk 4]
```diff
@@ -117,42 +117,41 @@ def get_links_between_two_nodes(from_node_id, to_node_id):
...
-                INNER JOIN here.routing_streets_name on edge = id'''
+					results.link_dir,
+					results.seq,
+					seg_lookup.segment_id,
+					ST_AsGeoJSON(streets.geom) AS geom,
+					ST_Length( ST_Transform(streets.geom,2952) ) / 1000 AS length_km
+				FROM results
+				JOIN here.routing_streets_22_2 AS streets USING ( link_dir )
+				JOIN congestion.network_links_22_2 AS seg_lookup USING ( link_dir )
+				ORDER BY seq;
```
Comment (Nate-Wessel) [inline backend/app/routes.py]: Indented with tabs. Please use spaces.

File: backend/app/routes.py [hunk 5]
```diff
@@ -117,42 +117,41 @@ def get_links_between_two_nodes(from_node_id, to_node_id):
...
 
     shortest_link_data = {
-        "source": source, 
-        "target": target,
-        "path_name": ', '.join(uniqueNames), 
-        "link_dirs": link_dirs, 
-        "geometry": json.loads(geometry) # parse json to object here; it will be dumped back to text in a second
+        "source": from_node_id, 
+        "target": to_node_id,
+        #"path_name": ', '.join(uniqueNames),
```
Comment (Nate-Wessel) [inline backend/app/routes.py]: Ditto, above. Can remove this line - we can do this on the frontend.

**Review by Nate-Wessel — CHANGES_REQUESTED** [review]
just a few little things; see comments

File: backend/app/routes.py (hunk 3 above)
Comment (Nate-Wessel) [reply backend/app/routes.py]: Actually... I think I'm changing my mind on this. We'll still need this on the backend to include in the output file. Feel free to leave this comment here.

File: backend/app/routes.py (hunk 4 above)
Comment (tankedman) [reply backend/app/routes.py]: Found the problem - when I copypasted back from pgadmin, it kept tab chars instead of spaces!

Comment (Nate-Wessel) [reply backend/app/routes.py]: You may also want to configure the tabbing/spacing options in PGAdmin. 

File -> Preferences -> Query Tool -> SQL Formatting

**Comment by Nate-Wessel** [comment]
@tankedman same here, it would be great to have some description/overview of what these changes are. I'm having trouble remembering after the long weekend.

**Pull request merged**